src/components/BoxBackground/BoxBackground.test.js

```diff
@@ -1,70 +1,99 @@
+import '@testing-library/jest-dom'
 
-const props = {
+const initialProps = {
   children: 'children',
   variant: 'info'
 }
 
-const component = <BoxBackground {...props} />
-
 describe('<BoxBackground /> mount', () => {
-  let wrapper
+  const { children } = initialProps
 
-  beforeEach(() => {
-    wrapper = mount(component)
-  })
+  it('is mounted', () => {
+    const { getByText } = render(<BoxBackground {...initialProps} />)
 
-  afterEach(() => {
-    wrapper.unmount()
-  })
+    const boxBackgroundNode = getByText(children)
 
-  it('is mounted', () => {
-    expect(wrapper.exists()).toBe(true)
+    expect(boxBackgroundNode).toBeVisible()
   })
 
-  it('has `boxBackground` class', () => {
-    expect(wrapper.children().hasClass('boxBackground')).toBe(true)
+  it(`should contains boxBackground class `, () => {
+    const { getByText } = render(<BoxBackground {...initialProps} />)
+
+    const boxBackgroundNode = getByText(children)
+
+    expect(boxBackgroundNode).toHaveClass('boxBackground')
   })
 
   it('has proper class name on success variant', () => {
-    wrapper.setProps({
+    const newProps = {
+      ...initialProps,
       variant: 'success'
-    })
+    }
+
+    const { variant } = newProps
+
+    const { getByText } = render(<BoxBackground {...newProps} />)
 
-    expect(wrapper.children().hasClass('boxBackground--success')).toBe(true)
+    const boxBackgroundNode = getByText(children)
+
+    expect(boxBackgroundNode).toHaveClass(`boxBackground--${variant}`)
   })
 
   it('has proper class name on warning variant', () => {
-    wrapper.setProps({
+    const newProps = {
+      ...initialProps,
       variant: 'warning'
-    })
+    }
+
+    const { variant } = newProps
 
-    expect(wrapper.children().hasClass('boxBackground--warning')).toBe(true)
+    const { getByText } = render(<BoxBackground {...newProps} />)
+
+    const boxBackgroundNode = getByText(children)
+
+    expect(boxBackgroundNode).toHaveClass(`boxBackground--${variant}`)
   })
 
   it('has proper class name on alert variant', () => {
-    wrapper.setProps({
+    const newProps = {
+      ...initialProps,
       variant: 'alert'
-    })
+    }
+
+    const { variant } = newProps
+
+    const { getByText } = render(<BoxBackground {...newProps} />)
+
+    const boxBackgroundNode = getByText(children)
 
-    expect(wrapper.children().hasClass('boxBackground--alert')).toBe(true)
+    expect(boxBackgroundNode).toHaveClass(`boxBackground--${variant}`)
   })
 
   it('has proper class name on progress variant', () => {
-    wrapper.setProps({
+    const newProps = {
+      ...initialProps,
       variant: 'progress'
-    })
+    }
 
-    expect(wrapper.children().hasClass('boxBackground--progress')).toBe(true)
+    const { variant } = newProps
+
+    const { getByText } = render(<BoxBackground {...newProps} />)
+
+    const boxBackgroundNode = getByText(children)
+
+    expect(boxBackgroundNode).toHaveClass(`boxBackground--${variant}`)
```

Comment:
a jak sobie w variancie wpiszę 'xd' to test też przejdzie

Reply:
Ok zrobiłem obiekt z variantami i sprawdzam dla każdego z variantów czy ma właściwą klasę